src/io/ipc/write/common.rs

```diff
@@ -379,3 +380,55 @@ pub struct EncodedData {
 pub(crate) fn pad_to_8(len: usize) -> usize {
     (((len + 7) & !7) - len) as usize
 }
+
+/// An array [`Chunk`] with optional accompanying IPC fields.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Record<'a> {
+    columns: Cow<'a, Chunk<Arc<dyn Array>>>,
```

Comment:
Curious: what is the purpose of `Cow`?

Reply:
The idea is to allow passing the arguments either by value or by reference. Obviously, better to avoid unnecessary clones if possible, but also wanted to avoid requiring a reference since then if you want to pass owned values you have to use the somewhat non-ergonomic `sink.feed((&chunk).into())` or `sink.feed((&chunk, &fields[..]).into())`. In contrast, with the chosen implementation you can use `sink.feed(chunk.into())` or `sink.feed((chunk, fields).into())` in all cases. 

I originally used a generic implementation that parameterized `Record` to accept a type implementing `Borrow<_>`, but it turns out that having a single type that implements sink for multiple other types (e.g. `FileSink` implementing both `Sink<Record<Chunk<_>>>` and `Sink<Record&<Chunk<_>>>`) doesn't work because the compiler can't infer the correct generic type when you call `sink.flush()` or `sink.close()`.

Reply:
Thank you for the explanation. Cool tip!